File: lua/Randwave.lua
local waveLengthBack = 0
local waveLengthForward = math.random(0, 15)

local function sleep(n)
   local runTime = os.clock()

   while os.clock() - runTime <= n do --[[ nothing ]] end
end

local function spaces(n)
   local i = 0

   while i <= n do
      io.write(" ")
      i = i + 1
   end
end

while true do
   
   local x = waveLengthBack
   while x < waveLengthForward do
      spaces(x)
      print("*")
      x = x + 1
      sleep(0.1)
   end

   waveLengthBack = math.random(0, 15)
   while not (waveLengthBack < waveLengthForward) do waveLengthBack = math.random(0, 15) end

   local y = waveLengthForward
   while y > waveLengthBack do
      spaces(y)
      print("*")
      y = y - 1
      sleep(0.1)
   end

   waveLengthForward = math.random(0, 15)
   while not (waveLengthBack < waveLengthForward) do waveLengthForward = math.random(0, 15) end
   
end

File: rust/Randwave.rs
use rand::Rng;
use std::time;

fn spaces(n: i8) {
    let mut i: i8 = 0;

    while i <= n {
	print!(" ");
	i+=1;
    }
}

fn main() {
    let mut rng = rand::thread_rng();

    let mut waveLengthBack: i8 = 0;
    let mut waveLengthForward: i8 = (rng.gen::<i8>()%16).abs();
    
    loop {
	let mut x = waveLengthBack;
	while x < waveLengthForward {
	    spaces(x);
	    println!("*");
	    x+=1;
	    std::thread::sleep(time::Duration::from_millis(65));
	}

	waveLengthBack = (rng.gen::<i8>()%16).abs();
	while !(waveLengthBack <= waveLengthForward) { waveLengthBack = (rng.gen::<i8>()%16).abs(); }

	let mut y = waveLengthForward;
	while y > waveLengthBack {
	    spaces(y);
	    println!("*");
	    y-=1;
	    std::thread::sleep(time::Duration::from_millis(65));
	}

	waveLengthForward = (rng.gen::<i8>()%16).abs();
	while !(waveLengthBack <= waveLengthForward) { waveLengthForward = (rng.gen::<i8>()%16).abs(); }

    }
}
